pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -96,11 +97,31 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
-	count := 0
-	if m.hasParams() {
-		count = 1
-		for _, param := range m.Params {
-			count *= len(param.Value.ArrayVal)
+	// Explicit combinations in the Matrix
+	// Calculate based on length of include Names and return count
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
+	paramNames := make(map[string][]string)
```

Comment:
this is not a list of param names, it's a map of the param name and value, maybe you can rename this to be clearer (especially given that you use this map to get the values in line 116 below)

Reply:
also wondering if it'd have been cleaner if we had and used a `params.extractParamMap` similar to  `params.extractParamValuesFromParams`:

https://github.com/tektoncd/pipeline/blob/771fb665f2df559b5ad2de111fea7bf2ff320881/pkg/apis/pipeline/v1/param_types.go#L125-L135

(sidenote: I think we should rename `params.extractParamValuesFromParams` to `params.extractValues` - cc @Yongxuanzhang @pritidesai)

Reply:
Ah that makes sense! it's redundant to say "FromParams" in the function name

Reply:
I have added a `params.extractParamMapArrVals` and `params.extractParamMapStringVals` helper function. I realize only `params.extractParamMapArrVals` is being used in this function, but in the implementation will use `params.extractParamMapStringVals` to map include params.

I also renamed `params.extractParamValuesFromParams` to`params.extractValues`.

cc: @jerop  @Yongxuanzhang, @pritidesai

Reply:
Oh okay, I reverted back